caikit/runtime/client/remote_module_base.py

```diff
@@ -0,0 +1,496 @@
+                    return {
+                        key: _remove_null_values(value)
+                        for key, value in _attr.items()
+                        if value
+                    }
+                if isinstance(_attr, list):
+                    return [
+                        _remove_null_values(listitem) for listitem in _attr if listitem
+                    ]
+
+                return _attr
+
+            http_request_dict = _remove_null_values(request_dm_dict)
+
+        # Get request options and target
+        request_kwargs = {"headers": {"Content-type": "application/json"}}
+        if self._tls.enabled:
+            if self._tls.cert_file and self._tls.key_file:
+                request_kwargs["cert"] = (
+                    self._tls.cert_file,
+                    self._tls.key_file,
+                )
+
+            if self._tls.insecure_verify:
+                request_kwargs["verify"] = False
+            else:
+                request_kwargs["verify"] = self._tls.ca_file or True
+
+        if method.output_streaming:
+            request_kwargs["stream"] = True
+
+        if self._connection.timeout:
+            request_kwargs["timeout"] = self._connection.timeout
+
+        request_kwargs = merge_configs(
+            request_kwargs, overrides=self._connection.options
+        )
+
+        request_url = (
+            f"{self._get_remote_target()}{get_http_route_name(method.rpc_name)}"
+        )
+
+        # Send request
+        response = requests.post(request_url, json=http_request_dict, **request_kwargs)
+        if response.status_code != 200:
+            raise CaikitCoreException(
+                CaikitCoreStatusCode.UNKNOWN,
+                f"Received status {response.status_code} from remote server: {response.text}",
+            )
+
+        # Parse response data model either as file or json
+        response_dm_class = DataBase.get_class_for_name(method.response_dm_name)
+
+        if method.output_streaming:
+
+            def stream_parser():
+                """Helper Generator to parse SSE events"""
+                for line in response.iter_lines():
+                    # Skip empty or event lines as they're constant
+                    if not line or b"event" in line:
+                        continue
+
+                    # Split data lines and remove data: tags before parsing by DM
+                    decoded_response = line.decode(response.encoding).replace(
+                        "data: ", ""
+                    )
+                    yield response_dm_class.from_json(decoded_response)
+
+            # Attach reference of this RemoteModuleClass to the returned DataStream. This ensures
+            # the GRPC Channel won't get closed until after the DataStream has been cleaned up
+            return_stream = DataStream(stream_parser)
+            return_stream._source = response.content
+            return return_stream
+
+        if response_dm_class.supports_file_operations:
```

Comment:
Iiiiinteresting. I know this is some new stuff you added, so it would be cool to see this spelled out in a comment